### PR TITLE
i16: new package with Interleave2/Deinterleave2 SIMD kernels (#79)

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Adds a new `i16` package providing `Interleave2`/`Deinterleave2` on `[]int16`, for splitting and packing raw 16-bit interleaved stereo PCM before widening to int32. This is PR2 of the integer-SIMD roadmap in #79 and mirrors the `i32` package (PR1, #81).

Interleaving is pure 16-bit-lane data movement (no arithmetic), so every kernel preserves the exact bit pattern of each lane, including sign and high bits.

**AMD64: tiered dispatch AVX2 (16 int16/iter) > SSE2 (8/iter) > pure Go.**
- Interleave: `VPUNPCKLWD`/`VPUNPCKHWD` + `VPERM2I128` (AVX2), `PUNPCKLWL`/`PUNPCKHWL` (SSE2).
- Deinterleave: `VPSHUFB` even/odd gather mask + `VPUNPCKLQDQ`/`VPUNPCKHQDQ` + `VPERMQ` (AVX2), `PSHUFLW`/`PSHUFHW`/`PSHUFD` + `PUNPCKLQDQ`/`PUNPCKHQDQ` (SSE2, no data mask).

**ARM64: NEON** `ZIP1`/`ZIP2` and `UZP1`/`UZP2` on `.8H`, hand-encoded as `WORD` with decoded comments that asmcheck cross-checks against `arm64asm`.

Pure-Go reference is the source of truth and the fallback. All paths are zero-allocation, clamp inputs to the shortest buffer, and use 16-bit scalar tails (`MOVW` on amd64, `MOVH` on arm64).

## Related Issues

Relates to #79 (PR2 of the i32/i16 integer-SIMD roadmap).

## Benchmarks

1000 int16 elements, zero-alloc, vs the pure-Go baseline:

| Op | amd64 AVX2 | arm64 NEON (Pi 5) |
|----|-----------|-------------------|
| Interleave2 | 71.9 ns, 13.2x | 164.8 ns, 12.8x |
| Deinterleave2 | 58.2 ns, 11.2x | 165.1 ns, 12.8x |

## Test Plan

- [x] Bit-exact SIMD-vs-Go parity (AVX2, SSE2, NEON) with sign/high-bit and int16-extreme patterns across sizes straddling every block width.
- [x] Scalar-tail no-overwrite guards for both output buffers.
- [x] Round-trip, length clamping (including odd-length ragged tails), empty/nil input, zero-alloc assertions.
- [x] `go vet`, full module suite, `golangci-lint run ./i16/...` (0 issues) on amd64.
- [x] asmcheck WORD decode with `SIMD_REQUIRE_OBJDUMP=1`; no goroutine-register clobber.
- [x] Verified on amd64 (AVX2) natively and arm64 (NEON) on a Raspberry Pi 5.

## New Features
- Added stereo PCM audio processing package with `Interleave2` and `Deinterleave2` functions for `int16` samples
- Optimized implementations with hardware-accelerated paths and efficient fallbacks across x86-64 and ARM64 architectures
- Zero-allocation operations designed for high-performance audio stream processing

## Tests
- Comprehensive test suite including correctness checks, performance benchmarks, boundary condition validation, and round-trip verification

### CODE REVIEW

File: i16/i16_arm64.s [hunk 1]
```diff
@@ -0,0 +1,93 @@
+//go:build arm64
+
+#include "textflag.h"
+
+// int16 (de)interleave on ARM64 (NEON / ASIMD).
+//
+// ZIP1/ZIP2/UZP1/UZP2 on .8H operands and the VLD1/VST1 .H8 loads/stores all
+// move 16-bit lanes by bit pattern, so these kernels mirror the int32 .4S
+// kernels in ../i32/i32_arm64.s with the lane width halved (8 lanes per 128-bit
+// register instead of 4); only the scalar tails differ (16-bit MOVW). The
```
[the remark this answers is not in the document]
Comment (tphakala) [reply i16/i16_arm64.s]: Fixed in 182ee02. The header comment now says 16-bit MOVH to match the scalar-tail code (MOVW was carried over from the i32 port).

File: i16/benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+func BenchmarkInterleave2_1000(b *testing.B) {
+	a := make([]int16, benchN)
+	c := make([]int16, benchN)
+	dst := make([]int16, benchN*2)
+	for i := range a {
+		a[i] = int16(i)
+		c[i] = int16(i + benchN)
+	}
+	b.SetBytes(benchN * 2 * 3)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply i16/benchmark_test.go]: Fixed in 182ee02. SetBytes now reports 8*benchN bytes: a + b (benchN int16 each) plus the 2*benchN dst (interleave) / src (deinterleave). ns/op and the speedup ratio are unchanged; only MB/s moves.
